src/RuleSet/RuleSet.php

```diff
@@ -235,18 +229,31 @@ public function removeRule($searchPattern): void
                 }
             }
         } else {
-            foreach ($this->rules as $propertyName => $rules) {
-                // Either no search rule is given or the search rule matches the found rule exactly
-                // or the search rule ends in “-” and the found rule starts with the search rule or equals it
-                // (without the trailing dash).
-                if (
-                    $searchPattern === null || $propertyName === $searchPattern
-                    || (\strrpos($searchPattern, '-') === \strlen($searchPattern) - \strlen('-')
-                        && (\strpos($propertyName, $searchPattern) === 0
-                            || $propertyName === \substr($searchPattern, 0, -1)))
-                ) {
-                    unset($this->rules[$propertyName]);
-                }
+            $this->removeMatchingRules($searchPattern);
+        }
+    }
+
+    /**
+     * Removes rules by property name or search pattern.
+     *
+     * @param string|null $searchPattern
+     *        pattern to remove. If null, all rules are removed. If the pattern ends in a dash,
```

Comment:
As we're introducing new methods anyway, I'd suggest to drop the nullability and a dedicated method `removeAllRules` instead.